src/Symfony/Component/Env/Dotenv.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Env;
+
+use Symfony\Component\Env\Exception\FormatException;
+use Symfony\Component\Env\Exception\FormatExceptionContext;
+use Symfony\Component\Env\Exception\PathException;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ExceptionInterface as ProcessException;
+
+/**
+ * Manages .env files.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class Dotenv
+{
+    const VARNAME_REGEX = '[A-Z][A-Z0-9_]*';
+    const STATE_VARNAME = 0;
+    const STATE_VALUE = 1;
+
+    private $path;
+    private $cursor;
+    private $lineno;
+    private $data;
+    private $end;
+    private $state;
+    private $export;
+    private $values;
+
+    /**
+     * Loads one or several .env files.
+     *
+     * @param ...string A list of files to load
+     *
+     * @throws FormatException when a file has a syntax error
+     * @throws PathException   when a file does not exist or is not readable
+     */
+    public function load(/*...$paths*/)
+    {
+        // func_get_args() to be replaced by a variadic argument for Symfony 4.0
+        foreach (func_get_args() as $path) {
+            if (!is_readable($path)) {
+                throw new PathException($path);
+            }
+
+            $this->populate($this->parse(file_get_contents($path), $path));
+        }
+    }
+
+    /**
+     * Sets values as environment variables (via putenv, $_ENV, and $_SERVER).
+     *
+     * Note that existing environment variables are never overridden.
+     *
+     * @param array An array of env variables
+     */
+    public function populate($values)
+    {
+        foreach ($values as $name => $value) {
+            if (isset($_ENV[$name]) || false !== getenv($name)) {
```

Comment:
I just think that people may be confused if they use an env var name that is not part of `$_ENV`, but exists in `$_SERVER`. It would then still be overridden though that we claim in the docblock that it wouldn't.